app/uk.gov.hmrc.helptosaveproxy/connectors/NSIConnector.scala

```diff
@@ -57,12 +57,12 @@ class NSIConnectorImpl @Inject() (conf: Configuration, metrics: Metrics, pagerDu
 
   val httpProxy: WSHttpProxy = new WSHttpProxy(conf)
 
-  override def createAccount(userInfo: NSIUserInfo)(implicit hc: HeaderCarrier, ec: ExecutionContext): EitherT[Future, SubmissionFailure, SubmissionSuccess] = {
+  override def createAccount(userInfo: NSIUserInfo, correlationId: Option[UUID])(implicit hc: HeaderCarrier, ec: ExecutionContext): EitherT[Future, SubmissionFailure, SubmissionSuccess] = {
     import uk.gov.hmrc.helptosaveproxy.util.Toggles._
 
     val nino = userInfo.nino
 
-    logger.info(s"Trying to create an account using NSI endpoint $nsiCreateAccountUrl", nino)
+    logger.info(s"Trying to create an account using NSI endpoint $nsiCreateAccountUrl, correlationId: ${correlationId.getOrElse("n/a")}", nino)
```

Comment:
instead of repeating `correlationId: ${correlationId.getOrElse("n/a")}` it'd be neater and easier to refactor in the future if you factored it out into a separate method like this:
```scala
def prettyFormatCorrelationId(id: Option[String]): String = s"correlationId: ${id.getOrElse("n/a")}"
```
and then called it everywhere it's needed.

Reply:
Another option is to extends `LoggingOps` to accommodate an optional correlation ID being passed in and passing it into `logger.warn`, `logger.info` etc

Reply:
ok done.